### PR TITLE
refactor(init): code cleanup and better test coverage

### PR DESCRIPTION
**Pull request by bearomorphism**
## Checklist

- [x] I have read the [contributing guidelines](https://commitizen-tools.github.io/commitizen/contributing/)

### Code Changes

- [ ] Add test cases to all the changes you introduce
- [ ] Run `poetry all` locally to ensure this change passes linter check and tests
- [ ] Manually test the changes:
  - [ ] Verify the feature/bug fix works as expected in real-world scenarios
  - [ ] Test edge cases and error conditions
  - [ ] Ensure backward compatibility is maintained
  - [ ] Document any manual testing steps performed
- [ ] Update the documentation for the changes

### Documentation Changes

- [x] Run `poetry doc` locally to ensure the documentation pages renders correctly

### CODE REVIEW

File: tests/commands/test_init_command.py [hunk 1]
```diff
@@ -206,22 +204,22 @@ def test_pre_commit_config_without_cz_hook(_, default_choice, tmpdir, config):
...
+            p.write(yaml.safe_dump({"repos": [CZ_HOOK_CONFIG]}))
 
             commands.Init(config)()
             check_cz_config(default_choice)
             # check that config is not duplicated
-            check_pre_commit_config([cz_hook_config])
+            check_pre_commit_config([CZ_HOOK_CONFIG])
 
 
 class TestNoPreCommitInstalled:
```
Comment (bearomorphism) [inline tests/commands/test_init_command.py:225]: `test_pre_commit_not_installed` is not working as expected...

File: tests/commands/test_init_command.py [hunk 2]
```diff
@@ -36,14 +37,14 @@ def unsafe_ask(self):
...
-expected_config = (
+EXPECTED_CONFIG = (
     "[tool.commitizen]\n"
     'name = "cz_conventional_commits"\n'
     'tag_format = "$version"\n'
     'version_scheme = "semver"\n'
-    'version = "0.0.1"\n'
     "update_changelog_on_bump = true\n"
     "major_version_zero = true\n"
+    'version = "0.0.1"\n'
```
Comment (bearomorphism) [inline tests/commands/test_init_command.py:47]: Does the order of these configs matter here?

File: commitizen/commands/init.py [hunk 3]
```diff
@@ -113,194 +110,176 @@ def __call__(self):
...
 
-            if not tag_format:
-                tag_format = default_format
-        return tag_format
+        return tag_format or default_format
 
     def _ask_version_provider(self) -> str:
         """Ask for setting: version_provider"""
 
         OPTS = {
```
Comment (woile) [inline commitizen/commands/init.py:208]: Ideally, the `VersionProvider` ABC class should include `identifier` and `description`. Which would allow us to do something like

```sh
providers = metadata.entry_points(group=PROVIDER_ENTRYPOINT)

{provider.identifier: provider.description for provider in providers} 
```

This way, custom user made providers would also appear in the init functions.

---

Just to be clear, this is optional, I hardcoded them here because it would help 90% of the users and I didn't want to go further.

File: tests/commands/test_init_command.py (hunk 2 above)
Comment (woile) [reply tests/commands/test_init_command.py:47]: Nop

Comment (bearomorphism) [reply tests/commands/test_init_command.py:47]: Thanks.
